### PR TITLE
Provide its own Dockerfile by a Job

### PR DESCRIPTION
**Pull request by iszulcdeepsense**
### Changed
The `job_types` method's format for plugins has been changed
  to support providing Dockerfiles directly by the Jobs.
  See [Developing plugins guide](https://github.com/TheRacetrack/racetrack/blob/470-provide-dockerfile-from-a-job-repository/docs/development/developing-plugins.md#job_types).
  Previous format is still supported for backward compatibility.

### CODE REVIEW

File: racetrack_commons/racetrack_commons/deploy/job_type.py [hunk 1]
```diff
@@ -15,22 +15,29 @@
 logger = get_logger(__name__)
 
 
+@dataclass
+class JobTypeImageDef:
+    source: str  # either 'jobtype' or 'job' depending on location of the Dockerfile
```
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply racetrack_commons/racetrack_commons/deploy/job_type.py:20]: Right, I'll turn into an enum.

File: docs/development/plugins-job-types.md
Comment (anders314159) [inline docs/development/plugins-job-types.md:1]: We should probably mention that using base_image is deprecated - unless the base_image on L23 is different from the other base_images in the code.

Comment (iszulcdeepsense) [reply docs/development/plugins-job-types.md:1]: Good find, I'll remove this occurrence

**Review by anders314159 — APPROVED** [review]

**Comment by anders314159** [comment]
Looks good to me.

**Pull request merged**